src/i18n.jl

```diff
@@ -0,0 +1,58 @@
+export AbstractLanguage, set_language!
+abstract type AbstractLanguage end
+
+
+
+
+abstract type English  <: AbstractLanguage end
+struct EnglishUS <: English end
+
+# computational_thinking.jl
+hint_str(lang::Lang) where {Lang <: English} = "Hint"
+tip_str(lang::Lang) where {Lang <: English} = "Tip"
+protip_invite_str(lang::Lang) where {Lang <: English} = "Curious to learn more?"
+protip_boxlabel_str(lang::Lang) where {Lang <: English} = "Pro Tip"
+almost_str(lang::Lang) where {Lang <: English} = "Almost there!"
+warning_box_str(lang::Lang) where {Lang <: English} = "Warning:"
+danger_str(lang::Lang) where {Lang <: English} = "Beware!"
+still_missing_str(lang::Lang) where {Lang <: English} = "Missing Response"
+still_missing_text_str(lang::Lang) where {Lang <: English} = md"Replace `missing` with your answer."
+still_nothing_str(lang::Lang) where {Lang <: English} = "Here we go!"
+still_nothing_text_str(lang::Lang) where {Lang <: English} = md"Replace `nothing` with your answer."
+wrong_type_str(lang::Lang) where {Lang <: English} = "Type Error"
+wrong_type_text_str(lang::Lang) where {Lang <: English} = md"Check the type of your response."
+wrong_type_text_str(lang::Lang, var, type) where {Lang <: English} = md"The type of $var should be $type"
+func_not_defined_str(lang::Lang) where {Lang <: English} = "Oopsie!"
+func_not_defined_text_str(lang::Lang, func_name) where {Lang <: English} = md"Make sure that you define a function called **$(Markdown.Code(string(func_name)))**"
+var_not_defined_str(lang::Lang) where {Lang <: English} = "Oopsie!"
+var_not_defined_text_str(lang::Lang, variable_name) where {Lang <: English} = md"Make sure that you define a variable called **$(Markdown.Code(string(variable_name)))**"
```

Comment:
Here are the functions where lang is the first parameter (in case we want to keep lang as the last argument).

Reply:
Done!